src/fsharp/CheckExpressions.fs

```diff
@@ -1688,7 +1688,7 @@ let CombineSyntacticAndInferredArities g declKind rhsExpr prelimScheme =
                     // If the expression infers additional arguments then use those (this shouldn't happen, since the
                     // arity inference done on the syntactic form should give identical results)
                     | [], ais | ais, [] -> ais
-                    | (h1 :: t1), (h2 :: t2) -> choose h1 h2 :: loop t1 t2
+                    | h1 :: t1, h2 :: t2 -> choose h1 h2 :: loop t1 t2
```

Comment:
To be honest I never remember the precedence of `::` and I add parentheses because of this.  It's the only cases where I noticed this so far.  But it's cool, we can remove them